### PR TITLE
feat(versioning): part 1 Add ref_pointers hash 

### PR DESCRIPTION
**Pull request by oreenlivnicode**
Add utils.py to download.
Add insert_workflow_or_action_to_redis function to download.py.
Add ref_pointers hash into ops db and use it to index, it does not contain commit sha yet.

image

### CODE REVIEW

File: src/downloader/utils.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            github_url,
+        )
+        redis_db.insert_to_hash(
+            object_path,
+            Config.redis_is_public_hash_field_name,
+            is_public,
+        )
+
+
+def add_ref_pointer_to_redis(uses_path: str, processed_path: str):
```
Comment (alex-ilgayev) [inline src/downloader/utils.py:30]: Suggestion:
Instead of sending two full paths,
maybe it will send 4 parameters, with path, ref, path, digest
WDYT?

**Review by alex-ilgayev — APPROVED** [review]
Looking good!
left few comments

File: src/config/config.py [hunk 2]
```diff
@@ -144,7 +145,7 @@ class Config:
     clean_redis: bool = None
 
     # Redis Config Constants
-    redis_sets_db: int = REDIS_SETS_DB
+    redis_objects_ops_db: int = REDIS_OBJECTS_OPS_DB
```
[the remark this answers is not in the document]
Comment (oreenlivnicode) [reply src/config/config.py:148]: it is operations, it is the db containing the download and index histories, and  now the ref_pointrs too

File: src/config/config.py [hunk 3]
```diff
@@ -154,6 +155,7 @@ class Config:
     action_download_history_set: str = REDIS_ACTION_DOWNLOAD_HISTORY_SET
     workflow_index_history_set: str = REDIS_WORKFLOW_INDEX_HISTORY_SET
     action_index_history_set: str = REDIS_ACTION_INDEX_HISTORY_SET
+    ref_pointers_hash: str = REDIS_REF_POINTERS_HASH
```
[the remark this answers is not in the document]
Comment (oreenlivnicode) [reply src/config/config.py:158]: hash

File: src/downloader/download.py [hunk 4]
```diff
@@ -102,23 +105,21 @@ def download_workflows_and_actions(repo: str) -> None:
...
-                    convert_raw_github_url_to_github_com_url(url),
-                )
-                workflows_db.insert_to_hash(
-                    workflow_unix_path,
-                    Config.redis_is_public_hash_field_name,
-                    is_public,
-                )
+            # Save workflow to redis
+            workflow_unix_path = convert_workflow_to_unix_path(repo, name)
+            github_url = convert_raw_github_url_to_github_com_url(url)
```
[the remark this answers is not in the document]
Comment (oreenlivnicode) [reply src/downloader/download.py:110]: Ok, it is for converting the raw.github URL into github contents URL the user will be able to click and open the file in it's specific version in the browser

File: src/downloader/utils.py (hunk 1 above)
Comment (oreenlivnicode) [reply src/downloader/utils.py:30]: How is it better? I have the two paths anyway, and it always comes after `insert_workflow_or_action_to_redis` function in which I use the "processed_path" to save the object

**Review by elad-pticha — CHANGES_REQUESTED** [review]
LGTM!

File: src/downloader/download.py [hunk 5]
```diff
@@ -102,23 +105,21 @@ def download_workflows_and_actions(repo: str) -> None:
...
-                    Config.redis_url_hash_field_name,
-                    convert_raw_github_url_to_github_com_url(url),
-                )
-                workflows_db.insert_to_hash(
-                    workflow_unix_path,
-                    Config.redis_is_public_hash_field_name,
-                    is_public,
-                )
+            # Save workflow to redis
+            workflow_unix_path = convert_workflow_to_unix_path(repo, name)
```
[the remark this answers is not in the document]
Comment (oreenlivnicode) [reply src/downloader/download.py:109]: We only use `convert_workflow_to_unix_path` in the `download_workflows_and_actions` function, where we iterate on every found workflow name in the repository, so we have to build a full path from it using the UNIX convert.
After that, in `download_action_or_reusable_workflow` function we receive the entire path from the API, so no need to convert it to UNIX path.

**Pull request closed without merging**